genesis/streaming.py

```diff
@@ -68,6 +68,20 @@ def __call__(self, msg):
 		for record in self.parser(val):
 			return topic, part, offs, self.filter(record)
 
+
+## message validation/processing
+
+def validate_and_process(msgs):
+	for msg in msgs:
+		if msg.error() is None:
+			# unpack and copy so we don't pickle the world (if multiprocessing)
+			yield (msg.topic(), msg.partition(), msg.offset(), msg.value())
+		elif msg.error().code() == KafkaError._PARTITION_EOF:
+			continue
+		else:
+			raise Exception(msg.error())
+
```

Comment:
Sounds great, I added this in. In the current scenario, I don't think the user would be able to test for the `_PARTITION_EOF` case since the filter passed in only gets information about the message's value, unless you meant further downstream in this client.